internal/registry/service.go

```diff
@@ -110,12 +111,7 @@ var (
 	}
 )
 
-func (client *Client) GetIteration(
-	ctx context.Context,
-	bucketSlug string,
-	opts ...GetIterationOption,
-) (*packer_service.PackerServiceGetIterationOK, error) {
-
+func (client *Client) GetIteration(ctx context.Context, bucketSlug string, opts ...GetIterationOption) (*models.HashicorpCloudPackerIteration, error) {
```

Comment:
I see that this now returns an iteration instead of the payload. I would check that all callers of this function have been updated to handle the new return value.

Reply:
disregard this comment - it was a comment to myself as I was reviewing.

Reply:
Seeing this, I wondered for a moment if the two methods should exist, but I couldn't find a reason why, so LGTM !

Reply:
@azr one gets a channel and the other gets the iteration. The get channel one is used in the `iteration` datasource and the other in the `image` datasource, and is also used before creating the iteration to check if it already exists!

Reply:
ah, nice !